### DISCUSSION TITLE
Display gameplay cursor when playing on touchscreen with stylus

### DISCUSSION THREAD

**Discussion by wizpizz** [ppy/osu#19331] [Ideas] [answered]
Unlike in osu!stable, in osu!lazer if you are using touchscreen, **the cursor almost completely disappears from the screen** (not 100% transparent). 

Although I understand why this is implemented, I think it would be also great to have the option to **toggle this feature** since there is currently no way to make your cursor visible while using touchscreen/stylus on the screen

Most touchscreen users would probably prefer no cursor anyway, but for people like me who like to use stylus on a touchscreen device, not seeing the cursor can be a disadvantage while playing if the beatmap background is completely dimmed out and the stylus is black

## Stable

![osu!_Sbk71XQMte](https://user-images.githubusercontent.com/76851476/180603644-792fe2fe-47ee-49c5-9d6d-043b2a3a07ff.gif)

## Lazer

![osu!_aYNzaGAtSa](https://user-images.githubusercontent.com/76851476/180603692-01f43ffb-f829-4396-b2d5-2150df10c15e.gif)

**Comment by peppy** [comment]
Can you explain why it's useful with a stylus? Are you not still directly touching the screen?

**Reply by wizpizz** [reply]
Yeah I am. What I mean by that is that the tip of the stylus easily blends in with the background (especially while playing fast beatmaps when you are moving stylus across the screen quickly), so being able to fully see the cursor makes it easier to keep track of the tip of the stylus.

Though I understand if this seems unnecessary since this targets a really specific group of people.

**Reply by Doksha** [reply]
I came to here specifically to request this, I use my touchscreen (usb) as a tablet for my main screen, so I am not actually using touchscreen functionality (sometimes I do though, depends on the day, but I dont plug the display in to my computer because I dont have a long enough vga cable +latency is too high), but there is no way to get the cursor to show, which makes it quite difficult to aim.
One advantage could also be the fact your hand is no longer obstructing the screen, while still having access to touchscreen functionality

Maybe a separate TD cursor skinning object, or a setting to adjust the opacity of the cursor when touchscreen is enabled

**Comment by wizpizz** [comment, marked as answer]
This has been implemented in the latest lazer update. Thanks peppy!
